### DISCUSSION TITLE
[@types/next-pwa] Next.js v13 is not supported

### DISCUSSION THREAD

**Discussion by std4453** [DefinitelyTyped/DefinitelyTyped#63843] [Issues with a @types package]
`@types/next-pwa` required Next.js v12:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fa1941eec2a45c2f7eaed18b9042294e17e77302/types/next-pwa/package.json#L4

While `next-pwa` supports Next.js above v9:

https://github.com/shadowwalker/next-pwa/blob/1e6af5fa9f6a807930f451adb5ab5078d5cda451/package.json#L29

If I have installed Next.js v13 in my project, installing `next-pwa` and `@types/next-pwa` would result in type mismatch with other plugins, say, `@nrwl/next/plugins/with-nx`, since they're referencing `NextConfig` from different versions of `next` package.

I believe the version requirement of `@types/next-pwa` can be changed to `^12.2.5 || ^13.0.0`, or, matching that of `next-pwa` itself, to `>=9.0.0`.

[a comment by an automated account is not shown]

**Comment by NivaldoFarias** [comment]
> While `next-pwa` supports Next.js above v9:
> 
> https://github.com/shadowwalker/next-pwa/blob/1e6af5fa9f6a807930f451adb5ab5078d5cda451/package.json#L29
> 
> If I have installed Next.js v13 in my project, installing `next-pwa` and `@types/next-pwa` would result in type mismatch with other plugins, say, `@nrwl/next/plugins/with-nx`, since they're referencing `NextConfig` from different versions of `next` package.
> 
> I believe the version requirement of `@types/next-pwa` can be changed to `^12.2.5 || ^13.0.0`, or, matching that of `next-pwa` itself, to `>=9.0.0`.

Nice catch! [Opened a PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63857) to resolve this issue, the changes should be up in a few days time.
